datashare-app/src/main/java/org/icij/datashare/tasks/TaskModifier.java

```diff
@@ -0,0 +1,6 @@
+package org.icij.datashare.tasks;
+
+@FunctionalInterface
+public interface TaskModifier {
```

Comment:
In my opinion the `TaskModifier` is not actually modifying the task, but it's rather publishing/broadcasting updates or event related to task changes.

For now it's only progress, but that could be anything that might be helpful to track tasks. The `TaskManager.getTask` will of course be the source of truth, but there could be other consumers of these events (say for instance we want to implement an event source pattern, then there will be a a component listening for these updates and record them somewhere).

I'm the neo4j implementation I have chosen the naming [TaskEvent](https://github.com/ICIJ/datashare-extension-neo4j/blob/main/neo4j-app/neo4j_app/icij_worker/task.py#L227) and [EventPublisher](https://github.com/ICIJ/datashare-extension-neo4j/blob/main/neo4j-app/neo4j_app/icij_worker/event_publisher/event_publisher.py#L6), but anything like conveying the idea of broadcasting/publishing an task related update/event would be appropriate in my opinion.